src/model/test/ThreeJSReverseTranslator_GTest.cpp

```diff
@@ -1179,4 +1185,75 @@ TEST_F(ModelFixture, ThreeJSReverseTranslator_FloorplanJS_OpenToBelow) {
   ASSERT_TRUE(infos[3].ceiling);
   EXPECT_FALSE(infos[3].ceiling->isAirWall());
   EXPECT_FALSE(infos[3].ceiling->adjacentSurface());
-}
\ No newline at end of file
+}
+
+TEST_F(ModelFixture, ThreeJSReverseTranslator_FloorplanJS_Site_ClimateZones_4166) {
+
+  // Test for #4166 - Merging FloorSpaceJS strips out climate zone assignment.
+
+  // Start with a base model, put some climate zone in there
+  // Add a Climate Zone to model1 only
```

Comment:
This test is a bit more involved than the ModelMerger test: it loads a floorspace.json and will also check that the mapping of things like north_axis actually ends up in the model.